### DISCUSSION TITLE
EN -> RU Translation

### DISCUSSION THREAD

**Discussion by ItsLeon15** [audiamus/AaxAudioConverter#103] [Ideas]
This isn't really an issue about the project but I was wondering if I could introduce Russian Language into the application?

Thanks

**Comment by audiamus** [comment]
Thank you very much for your suggestion. I guess you have done such a thing before. 

Are you familiar with the .Net environment, Windows Forms and its concept of multi-language support, resource files etc? And with Visual Studio?

(In the ideal case, the user manual would have to be translated, too. It's written with LibreOffice.)

You are also aware that translations need updates with almost every new version of the app? Because of that, I do not recommend to start translating at the moment because I am currently working on AAX Audio Converter 2 which shall add a direct download option for your books and also reshuffle much of the existing code.

> This isn't really an issue about the project 

So, I move it to the discussion section.

**Comment by ItsLeon15** [comment]
> I am currently working on AAX Audio Converter 2

Ah okay, I will wait until you release that version and then translate it. 

> Are you familiar with the .Net environment, Windows Forms and its concept of multi-language support, resource files etc? And with Visual Studio?

Yes I am. I understand that .Net environment allows mutli-language support and is one of the reasons I would like to contribute with this project. 

> You are also aware that translations need updates with almost every new version of the app?

I am aware that translations will need to be updated, I understand that and am willing to help as much as I can.

**Comment by audiamus** [comment]
Very good. 

You may have seen that I tried to keep all language resources in UI assemblies and I intend to continue so. There will be an extra UI lib in the future where I put all the new stuff. The property grid adapter of the main window, which currently has its own resource file, will not longer require that. With more reflection tricks and attribute usage I think I can move its resources to the assembly resource file. You may have noticed the rather complex mechanism to find the language resources for the property grid adapter elements. Complex, but it does the job. 

AAX Audio Converter 2 will migrate to the .Net 5 platform, not to .Net 6 initially, so you can still use VS2019.
